### DISCUSSION TITLE
computed properties presist is not persisting

### DISCUSSION THREAD

**Discussion by sohailahmad07** [livewire/livewire#6594] [7. Bugs (archived)]
### Livewire version

3.0.1

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

I am using a form where i get all of the categories for the users and i am also using live validation but when it render the component due to live modifier it re-ran the query 

1. Create a livewire component 
2. Add a computed property with persist set to true
3. Add an input on the page with wire:modal.live 
4. When you start typing into the input and i re-render components it will also re-query the computed property 

### Additional information

components 
```
 public string $title = '';
 
#[Computed(persist: true)]
    public function categories(): Collection
    {
        returnCategory::all(['id', 'name']);
    }
```

view
```
 <input type="text" name="title" wire:model.live.debounce.500ms="title"   />
 
  <select name="category" wire:model.live.debounce="category">
                        <option value="null" disabled>Select Category</option>
                        @foreach($this->categories() as $categoryItem)
                            <option value="{{ $categoryItem['id'] }}">{{ $categoryItem['name'] }}</option>
                        @endforeach
                    </select>
```

you can use debugbar to monitor the query 

look at the last query time you will see it change when i stop typing 

https://imgur.com/a/BFpzOVl

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by xteron** [comment]
Stumbled over similar problem. This solved it for me.

Change 
`@foreach($this->categories() as $categoryItem)`
To
`@foreach($this->categories as $categoryItem)`

Remove () when accessing categories.
